python_modules/libraries/dagster-components/dagster_components/resolved/model.py

```diff
@@ -284,14 +304,22 @@ def derive_field_resolver(annotation: Any, field_name: str) -> "Resolver":
     )
 
 
+def _recurse(context: "ResolutionContext", field_value):
+    return context.resolve_value(field_value)
+
+
 class Resolver:
     """Contains information on how to resolve a field from a ResolvableModel."""
 
     def __init__(
         self,
         fn: Union[ParentFn, AttrWithContextFn, Callable[["ResolutionContext", Any], Any]],
+        model_field_name: Optional[str] = None,
```

Comment:
Is any combo of `None` valid here?

Reply:
can you clarify?

Reply:
There are four states here of "None-ness"  

* Both None
* Neither None
* only field is None
* only type is None

Are all four valid?

Reply:
yep all valid